swift_cells.py

```diff
@@ -206,6 +211,21 @@ def __init__(
                 critical_density, units=internal_density_unit
             )
 
+            # Read in the softening lengths, determine whether to use comoving or physical
+            self.dark_matter_softening = min(
+                float(self.parameters.get("Gravity:comoving_DM_softening", 0)) * self.a,
```

Comment:
Is there a use case where we need the softening to default to zero? Would it be better to index the dict here and let the KeyError terminate the program if the softening information isn't where we expect it?

Reply:
I've changed it to index the dict. I also realised that if someone did a run with comoving_baryon_softening set, but without max_physical_baryon_softening, then it would give the wrong value. I'm not sure how likely that is though